src/AutoRest.CSharp/Common/Output/Models/Types/ObjectType.cs

```diff
@@ -429,10 +429,10 @@ private ObjectTypeDiscriminatorImplementation[] CreateDiscriminatorImplementatio
             )).ToArray();
         }
 
-        private IEnumerable<ObjectTypeProperty> BuildProperties()
+        protected IEnumerable<ObjectTypeProperty> BuildProperties(bool includeParent = true)
         {
             // WORKAROUND: https://github.com/Azure/autorest.modelerfour/issues/261
-            var existingProperties = GetParentProperties();
+            var existingProperties = includeParent ? GetParentProperties() : new HashSet<string?>();
```

Comment:
The parameter name seems to be wrong and I'm not sure if you need to make this change. This HashSet is used to remove parent properties that are already defined in the base class.

Reply:
It refers to whether or not you want to include the parent properties (or base properties) and is used by this
        public ObjectTypeProperty[] MyProperties => _myProperties ??= BuildProperties(false).ToArray();
which is a new property on MgmtObjectType used when we only want to see the properties that the current type explicitly adds

Reply:
This is not how the HashSet is used though. It's used to skip properties in the following line.

```
                    if (existingProperties.Contains(property.Language.Default.Name))
```

We don't use it to add more properties to the type.